apps/scan/backend/src/app_flow.ts

```diff
@@ -0,0 +1,155 @@
+import { InsertedSmartCardAuthApi } from '@votingworks/auth';
+import { doesUsbDriveRequireCastVoteRecordSync } from '@votingworks/backend';
+import { assert, throwIllegalValue } from '@votingworks/basics';
+import { PrecinctScannerState } from '@votingworks/types';
+import { UsbDrive } from '@votingworks/usb-drive';
+import {
+  isElectionManagerAuth,
+  isPollWorkerAuth,
+  isSystemAdministratorAuth,
+} from '@votingworks/utils';
+import { BALLOT_BAG_CAPACITY } from './globals';
+import { Store } from './store';
+import { constructAuthMachineState } from './util/construct_auth_machine_state';
+import { AppFlowState } from './types';
+
+export async function getCurrentAppFlowState({
```

Comment:
The structure of this function was taken from the frontend's `AppRoot`. Instead of returning e.g. `<SetupCardReaderScreen />` I'm returning an equivalent status of `'setup_card_reader'`.

Reply:
Is the vision that this function will drive the frontend eventually?

Reply:
As discussed elsewhere in this PR, that was my initial goal. I've retreated from that with the subsequent changes.